src/ws/kms_window_system.cpp

```diff
@@ -251,7 +251,7 @@ ManagedResource<gbm_device*> create_gbm_device(int drm_fd)
 
 ManagedResource<int> open_active_vt()
 {
-    auto fd = open("/dev/tty0", O_RDONLY);
+    auto fd = open("/dev/tty", O_RDONLY);
```

Comment:
Not sure about the situation on FreeBSD, but on Linux /dev/tty0 refers to the active VT, whereas /dev/tty refers to the tty of the process (group), which may not be an actual VT. For example, when running vkmark through SSH /dev/tty would point to a pseudo-terminal.

Reply:
Fixed. I was mainly concerned with hardcoding `/dev/tty0` when user maybe on `/dev/tty1` or similar. Opening active terminal is kinda pointless given you can just use stdin and bail out if any `VT_*` ioctl fail.

Reply:
I have created an alternative MR here: https://github.com/vkmark/vkmark/tree/unbreak-freebsd. That MR first tries `/dev/tty` and then `/dev/tty0` (note that `/dev/tty0` is special on linux, and refers to the currently active VT, whatever that may be). It also uses `SIGINT` for all the `VT_PROCESS` handling, so that we just stop `vkmark` if any VT related request is made.

Let me know if that MR would work OK for FreeBSD. Thanks!

Reply:
[unbreak-freebsd](https://github.com/vkmark/vkmark/tree/unbreak-freebsd) branch works here as good as this PR. I'd still prefer stdin over `/dev/tty`, so users can redirect stdin to a proper VT when running under `tmux` or `seatd`.

```
$ doas pkg install seatd
$ doas service seatd onestart
# Check /dev/ttyv0 isn't open yet as no Wayland compositors are running
$ doas procstat -f $(pgrep seatd)
  PID COMM                FD T V FLAGS    REF  OFFSET PRO NAME        
 1234 seatd             text v r r-------   -       - -   /usr/local/bin/seatd
 1234 seatd              cwd v d r-------   -       - -   /                 
 1234 seatd             root v d r-------   -       - -   /                 
 1234 seatd                0 v x rw------   4     223 -   -                 
 1234 seatd                1 p - rw------   2       0 -   -                 
 1234 seatd                2 p - rw------   2       0 -   -                 
 1234 seatd                3 p - rw---n--   1       0 -   -                 
 1234 seatd                4 p - rw---n--   1       0 -   -                 
 1234 seatd                5 s - rw---n--   1       0 UDS 0 0 /var/run/seatd.sock
$ vidcontrol -s 1 </dev/ttyv0 # aka "chvt 1" or Ctrl+Alt+F1

$ vkmark --winsys kms
Error: Failed to open active VT
$ vkmark --winsys kms </dev/ttyv1
=======================================================
    vkmark 2017.08-38-g8a70e40
=======================================================
    Vendor ID:      0x8086
    Device ID:      0x1912
    Device Name:    Intel(R) HD Graphics 530 (SKL GT2)
    Driver Version: 96473187
    Device UUID:    f90ad5f8b8bbf79c6fbb467bddb9f37e
=======================================================
[vertex] device-local=true: FPS: 60 FrameTime: 16.667 ms
[vertex] device-local=false: FPS: 59 FrameTime: 16.949 ms
^C
```

```diff
--- a/src/ws/kms_window_system.cpp	(8a70e40d7b82)
+++ b/src/ws/kms_window_system.cpp	(working copy)
@@ -252,7 +252,9 @@ int open_vt(const char *path)
 int open_vt(const char *path)
 {
     struct vt_mode vt_mode;
-    auto fd = open(path, O_RDONLY);
+    auto fd = 0;
+    if (path)
+        auto fd = open(path, O_RDONLY);
 
     if (fd < 0)
         return -1;
@@ -268,10 +270,10 @@ ManagedResource<int> open_active_vt()
 
 ManagedResource<int> open_active_vt()
 {
-    auto fd = open_vt("/dev/tty");
+    auto fd = open_vt(NULL);
     if (fd < 0)
     {
-        Log::debug("/dev/tty is not a VT, trying to use /dev/tty0\n");
+        Log::debug("stdin is not a VT, trying to use /dev/tty0\n");
         fd = open_vt("/dev/tty0");
         if (fd < 0)
             throw std::runtime_error{"Failed to open active VT"};
```

Reply:
> note that `/dev/tty0` is special on linux, and refers to the currently active VT, whatever that may be

On DragonFly and FreeBSD to get active VT one has to issue `VT_GETACTIVE` against `/dev/ttyv0` (or stdin, to let user override). NetBSD (`/dev/ttyE0`) and OpenBSD (`/dev/ttyC0`) also support `VT_GETSTATE`, similar to Linux.

Reply:
Err, I'm not suggesting to complicated the code with `VT_GETACTIVE`. Making `vkmark` use physical console while connected over SSH can be surprising. For example, Wayland apps don't to try to use `wayland-0` when `WAYLAND_DISPLAY` is unset or X11 app don't try to use `:0` when `DISPLAY` is unset.

Reply:
I pushed another update that allows the user to specify the TTY to use, but as a winsys option, instead of indirectly via stdin, e.g., `vkmark --winsys=kms --winsys-options=kms-tty=/dev/ttyv1`. I would prefer not to hardcode an association between stdin and TTY, in case we want to pass other information through stdin in the future. Let me know how this works for you. Thanks!